packages/reports/src/Components/Form/CompositeCalendar/two-month-picker.tsx

```diff
@@ -1,63 +1,86 @@
+    );
     const [right_pane_date, setRightPaneDate] = React.useState(value);
 
-    const navigateFrom = e => {
-        setLeftPaneDate(e.unix());
-        setRightPaneDate(addMonths(e, 1).unix());
+    /**
+     * Navigate from date
+     *
+     * @param {moment.Moment} date
+     * @returns {void}
+     */
+    const navigateFrom = (date: moment.Moment): void => {
+        setLeftPaneDate(date.unix());
+        setRightPaneDate(addMonths(date.toISOString(), 1).unix());
     };
 
     /**
-     * Only allow previous months to be available to navigate. Disable other periods
+     * Navigate to date
      *
-     * @param date
-     * @param range
-     * @returns {boolean}
+     * @param {moment.Moment} date
+     * @returns {void}
      */
-    const validateFromArrows = (date, range) => {
-        return range === 'year' || diffInMonths(epochToMoment(left_pane_date), date) !== -1;
+    const navigateTo = (date: moment.Moment): void => {
+        setLeftPaneDate(subMonths(date.toISOString(), 1).unix());
+        setRightPaneDate(toMoment(date).unix());
     };
 
     /**
-     * Validate values to be date_from < date_to
+     * Only allow previous months to be available to navigate. Disable other periods
+     *
+     * @param {moment.Moment} date
+     * @param {Extract<moment.DurationInputArg2, 'month'>} range
+     * @returns {boolean}
      */
-    const shouldDisableDate = date => {
-        return isPeriodDisabled(date.unix());
+    const validateFromArrows = (date: moment.Moment, range: Extract<moment.DurationInputArg2, 'month'>): boolean => {
+        // return range === 'year' || diffInMonths(epochToMoment(left_pane_date), date) !== -1;
+        return diffInMonths(epochToMoment(left_pane_date), date) !== -1;
     };
 
     /**
      * Only allow next month to be available to navigate (unless next month is in the future).
      * Disable other periods
      *
-     * @param date
-     * @param range
+     * @param {moment.Moment} date
+     * @param {Extract<moment.DurationInputArg2, 'month'>} range
      * @returns {boolean}
      */
-    const validateToArrows = (date, range) => {
-        if (range === 'year') return true; // disallow year arrows
+    const validateToArrows = (date: moment.Moment, range: Extract<moment.DurationInputArg2, 'month'>): boolean => {
+        // if (range === 'year') return true; // disallow year arrows
         const r_date = epochToMoment(right_pane_date).startOf('month');
         if (diffInMonths(toMoment().startOf('month'), r_date) === 0) return true; // future months are disallowed
         return diffInMonths(r_date, date) !== 1;
     };
 
-    const jumpToCurrentMonth = () => {
+    /**
+     * Validate values to be date_from < date_to
+     *
+     * @param {moment.Moment} date
+     * @return {boolean}
+     */
+    const shouldDisableDate = (date: moment.Moment): boolean => {
+        return isPeriodDisabled(date.unix());
+    };
+
+    const jumpToCurrentMonth = (): void => {
```

Comment:
Do we need to define the return type here? I guess we can remove void type :thinking: